Cilsil/Program.cs

```diff
@@ -118,19 +116,32 @@ private static void Main(string[] args)
         /// Executes the translation.
         /// </summary>
         /// <param name="paths">The paths.</param>
-        /// <param name="printprocs">The printprocs.</param>
-        /// <param name="debug">The debug flag.</param>
-        /// <returns></returns>
-        public static (Cfg, TypeEnvironment) ExecuteTranslation(string[] paths,
-                                                                string printprocs = null,
-                                                                bool debug = false)
+        /// <param name="extensionProgress">If <c>true</c>, periodically write progress to 
+        /// console if the input binaries are adequately large.</param>
+        /// <returns>The computed cfg and type environment.</returns>
+        public static (Cfg, TypeEnvironment) ExecuteTranslation(
+            string[] paths, bool extensionProgress = false)
         {
             var assemblies = GetAssemblies(paths);
 
+            var totalSize = assemblies.Select(p =>
+            {
+                try
+                {
+                    return new FileInfo(p).Length;
+                }
+                catch
+                {
+                    return 0;
+                }
+            }).ToArray().Sum();
+
             InstructionParser.RegisterAllKnownParsers();
 
-            var decompilationService = new DecompilationService(assemblies);
-            var tenvParser = new TenvParserService();
+            var reportProgressExtension = totalSize > 1e7 && extensionProgress;
```

Comment:
Do we report progress bar only when we have more than 10000000 methods?

Reply:
no, only when total assembly bytes >1e7 (so that we don't spam progress information in the extension when the no. bytes is too small)